design/vela-core/workflow_policy.md

```diff
@@ -199,38 +201,112 @@ Here are the steps in Workflow Manager:
 
 - The Workflow Manager will reconcile the workflow step by step.
 - For each step, it will create a task for Task manager to handle.
-- It will then check the `observedAppGeneration` for this step.
-  This is to solve the [issue of passing data from the old generation][1].
-  - If the generation matches the current one, it will continue next step.
-  - Otherwise, it will retry reconcile later.
+- The `workflow.stepIndex`  is used to record the step of the workflow execution，it will increase automatically after each task is executed.
 
   An example of the status:
 
   ```yaml
   kind: Application
   metadata:
-    generation: 2
+    name: foo
   status:
+    phase: runningWorkflow
     workflow:
+      stepIndex: 1
       steps:
       - name: ...
-        observedAppGeneration: 1
   ```
 
 Here are the steps in Task Manager:
 
-- Workflow Manager will call Task Manager to enqueue a task.
-  Note that if a task with the same name exists, it will append new one.
-- Task Manager will have another go routine to pop a task from the queue to execute.
-- Each task is a CUE graph that must have `continue` field:
+- Workflow Manager will call Task Manager to run a task.
+- Task Manager execute all steps of the task in order.
+- When the `continue` step is encountered, and the `return` field is true, Task Manager will stop continuing to execute the task.
+
+  It will issue a delay reconcile event, in the next application reconcile，the task will be re-executed
 
   ```yaml
-  continue: vela.#Continue & {
+  continue: op.#Continue & {
     return: true # This will be used as the return of the CUE task
   }
   ```
 
-- If `continue.return == true`, Task Manager will update the `observedAppGeneration` for this step in status.
+
+### Workflow Operation
+
+#### Terminate Workflow
+
+If the execution of the workflow does not meet expectations, it may be necessary to terminate the workflow
+
+There are two ways to achieve that
```

Comment:
```suggestion
There are two ways to achieve that:
```

Reply:
The following two approaches are not mutual-exclusive. The op.Break can be captured and then report terminated status.